src/Command/InitializeAdherentEmailSubscriptionHistoryCommand.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace AppBundle\Command;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\AdherentEmailSubscriptionHistory;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Internal\Hydration\IterableResult;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class InitializeAdherentEmailSubscriptionHistoryCommand extends Command
+{
+    private const BATCH_SIZE = 50;
+
+    protected static $defaultName = 'app:adherent:initialize-email-subscriptions-history';
+
+    /**
+     * @var SymfonyStyle
+     */
+    private $io;
+
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Create email subscriptions history. The history will be created for both users and adherents even inactives.')
+        ;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($this->isAlreadyInitialize()) {
+            $this->io->error('Impossible to initialize email subscription history. It already exists.');
+
+            return 1;
+        }
+
+        $this->io->title('Starting email subscription history initialization.');
+
+        $progressBar = new ProgressBar($output, $this->getAdherentCount());
+
+        $this->em->beginTransaction();
+
+        $nb = 0;
+        foreach ($this->getAdherents() as $result) {
+            $adherent = reset($result);
```

Comment:
why not just read the first element `$result[0]`?

Reply:
I guess it is more accurate to use $result[0] in that case because $result does not have random keys.